packages/hardhat-core/src/internal/hardhat-network/provider/TxPool.ts

```diff
@@ -75,29 +75,79 @@ export class TxPool {
   constructor(
     private readonly _stateManager: PStateManager,
     blockGasLimit: BN,
+    minGasPrice: BN,
     common: Common
   ) {
     this._state = makePoolState({
       blockGasLimit: bnToHex(blockGasLimit),
+      minGasPrice: bnToHex(minGasPrice),
     });
     this._deserializeTransaction = (tx) => deserializeTransaction(tx, common);
   }
 
   public async addTransaction(tx: Transaction) {
     const senderAddress = this._getSenderAddress(tx);
-    const senderNonce = await this.getExecutableNonce(senderAddress);
+    const accountNonce = await this._getAccountNonce(senderAddress);
+    const nextNonce = await this.getNextNonce(senderAddress);
```

Comment:
I renamed `executableNonce` to `nextNonce`. I'm also using `accountNonce` as the nonce that only considers mined transactions. Please let me know if there's a better vocabulary to use for this.

Reply:
TBH I don't understand what each means now.

What about something like `nextPendingNonce` and `nextConfirmedNonce`?

Reply:
I like that idea, but let's wait until we merge as many PRs as we can, because this rename already caused a lot of conflicts with other PRs :sweat_smile: